link.go

```diff
@@ -561,11 +563,13 @@ func (xdp *LinkXDP) decode(ad *netlink.AttributeDecoder) error {
 }
 
 func (xdp *LinkXDP) encode(ae *netlink.AttributeEncoder) error {
-	ae.Uint32(unix.IFLA_XDP_FD, xdp.FD)
-	ae.Uint32(unix.IFLA_XDP_EXPECTED_FD, xdp.ExpectedFD)
-	ae.Uint8(unix.IFLA_XDP_ATTACHED, xdp.Attached)
+	ae.Int32(unix.IFLA_XDP_FD, xdp.FD)
+	ae.Int32(unix.IFLA_XDP_EXPECTED_FD, xdp.ExpectedFD)
 	ae.Uint32(unix.IFLA_XDP_FLAGS, xdp.Flags)
-	ae.Uint32(unix.IFLA_XDP_PROG_ID, xdp.ProgID)
-
+	// XDP_ATtACHED and XDP_PROG_ID are things that only can return from the kernel,
+	// not be send, so we don't encode them.
+	// source: https://elixir.bootlin.com/linux/v5.10.15/source/net/core/rtnetlink.c#L2894
+	// ae.Uint8(unix.IFLA_XDP_ATTACHED, xdp.Attached)
+	// ae.Uint32(unix.IFLA_XDP_PROG_ID, xdp.ProgID)
```

Comment:
If there's no need to encode them, there's probably no need to keep the lines here as commented code.